### ISSUE TITLE
Extra credentials in connection 

### ISSUE THREAD

**Issue by mahic** [starburstdata/dbt-trino#385]
### Describe the feature

dbt-trino doesn't support extra_credential while trino-python-client [supports it](https://github.com/trinodb/trino-python-client?tab=readme-ov-file#extra-credentials)

### Describe alternatives you've considered

Use http_headers to supply X-Trino-Extra-Credential

This doesn't work with multiple extra credentials because http_headers is a dict.

### Who will benefit?

Users who usually use extra credentials

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by mahic** [comment]
@damian3031 [ref comment](https://github.com/trinodb/trino-python-client/issues/6#issuecomment-833295159) seems like the workaround might be the following:

`http_headers: {X-Trino-Extra-Credential: "cred1=xxx,cred2=yyyy,cred3=zzzz"}` 

What threw us off was that when we tested this with DBeaver, the client sent 2 headers. This is because trino-client (and Trino JDBC Driver) seems to [prefer to send multiple headers](https://github.com/trinodb/trino/blob/f745b68ff2000159023926ad6c9b23be596182b0/client/trino-client/src/main/java/io/trino/client/StatementClientV1.java#L184) over the short-form

However, since trino-python-client supports the extra_credential parameter explicitly (and so does the trino-client/Trino JDBC driver), dbt-trino should match this IMO, do you agree?

**Comment by hovaesco** [comment]
> However, since trino-python-client supports the extra_credential parameter explicitly (and so does the trino-client/Trino JDBC driver), dbt-trino should match this IMO, do you agree?

Yes, it should match, feel free to post a PR with the implementation, we're ready to help.
